lib/src/mapbox_map.dart

```diff
@@ -675,3 +768,53 @@ class _GestureListener extends GestureListener {
     onMapScrollListener?.call(context);
   }
 }
+
+/// Listen for a single interaction added to the map, identified by its id
+class _InteractionListener<T extends FeaturesetFeature>
+    extends _InteractionsListener {
+  _InteractionListener({
+    required this.interactionID,
+    required this.onInteractionListener,
+  });
+
+  int interactionID;
+
+  final OnInteraction<T> onInteractionListener;
+
+  @override
+  void onInteraction(MapContentGestureContext context,
+      FeaturesetFeature feature, int interactionID) {
+    var featuresetID = feature.featureset.featuresetId;
+    T typedFeature;
+
+    if (featuresetID == "buildings") {
+      typedFeature =
+          StandardBuildingsFeature.fromFeaturesetFeature(feature) as T;
+    } else if (featuresetID == "poi") {
+      typedFeature = StandardPOIsFeature.fromFeaturesetFeature(feature) as T;
+    } else if (featuresetID == "place-labels") {
+      typedFeature =
+          StandardPlaceLabelsFeature.fromFeaturesetFeature(feature) as T;
+    } else {
+      typedFeature = feature as T;
+    }
+
+    onInteractionListener.call(context, typedFeature);
+  }
+}
+
+/// Listen to all interactions on the map, determine which interaction to call
+class _InteractionsList<T extends FeaturesetFeature>
```

Comment:
nit: _InteractionsMap or _InteractionsContainer would be a better name, since it is implemented as Map, not as List internally.